dashboard/app/models/script_level.rb

```diff
@@ -117,7 +117,7 @@ def self.add_script_levels(script, lesson_group, lesson, raw_script_levels, coun
 
       # Generate and store the seed_key, a unique identifier for this script level which should be stable across environments.
       # We'll use this in our new, JSON-based seeding process.
-      seed_context = ScriptSeed::SeedContext.new(script: script, lesson_groups: [lesson_group], lessons: [lesson])
+      seed_context = ScriptSeed::SeedContext.new(script: script, lesson_groups: [lesson_group], lessons: [lesson], lesson_activities: [], activity_sections: [])
```

Comment:
hmm... yes... I was thinking this would only be done for legacy scripts, because it is part of the old seed process. but since we are still using the old seed process on migrated scripts, this will have to change. 

I was thinking of just removing this block of code, since it is only needed as a sort of dry run for launching new script seeding for legacy scripts. but it seems like we might not be trying to do that any time soon, or we may never do it in the sense that we may just "migrate" these scripts instead, which would be a different way to get them onto the new script seeding system. How does that sound?

Reply:
I went ahead and removed this block of code in this PR, since I don't think we want to risk leaving things in a state where the old/current seed process running could mess up imported/migrated scripts.

Reply:
removing this broke some tests, most notably this makes it so that we don't have anything preventing us from adding the same level to a script twice:
```
===[0m[1000D[K FAIL["test_level_can_only_be_added_once_to_a_lesson", "ScriptTest", 969.5705375440011]
 test_level_can_only_be_added_once_to_a_lesson#ScriptTest (969.57s)
        StandardError expected but nothing was raised.
        test/models/script_test.rb:2656:in `block in <class:ScriptTest>'
        test/testing/setup_all_and_teardown_all.rb:36:in `run'

[0m[1000D[K FAIL["test_script_import_is_idempotent_w.r.t._positions_and_count", "ScriptTest", 971.1916625019949]
 test_script_import_is_idempotent_w.r.t._positions_and_count#ScriptTest (971.19s)
        Expected: 5
          Actual: 0
        test/models/script_test.rb:310:in `block in <class:ScriptTest>'
        test/testing/setup_all_and_teardown_all.rb:36:in `run'
```
I need to think on it a bit more to figure out what to do here.

Reply:
I think that the constraint of a level should be unique in a script makes sense to me

Reply:
^ agreed. When I wrote my original comment, I didn't realize this was the old seeding logic 🙈. So this makes a lot of sense and I think it's fine to add the constraint of a level being unique within a script.

Are there any stakeholders we should ask before adding this hard constraint? I think it's been kind of an assumption before but if it's going to be codified in our seeding logic, it seems like it would be hard to change in the future

Reply:
Thanks Bethany. General thoughts:
1. this PR doesn't add this constraint. it was added in October: https://github.com/code-dot-org/code-dot-org/pull/36984
2. when this constraint was added, there were only a couple of scripts which had duplicate levels. IIRC at most one of them were in live scripts. I believe they were all cloned and/or removed in this PR: https://github.com/code-dot-org/code-dot-org/commit/1cb2d28f4c33eb4c01a54acfb23ce99c9dfb3acf 

Thinking through our new lesson edit UI, I would think that for now we could show an error if they try to save a lesson that contains a duplicate level. that might be all we need to do for now (January Launch? January ASAP?) to keep bad data from getting into our system. Opened here: https://codedotorg.atlassian.net/browse/PLAT-612

 in the future, I could see graying out the `+` button in the add level dialog to make this a bit more user-friendly. I'm not sure if this needs to be tracked -- we could also wait and see if anyone runs into this before opening a ticket for it.

Reply:
@bethanyaconnor you make a good point about considering stakeholders needs as we commit harder in this direction. I think our long term vision is to get closer to levels appearing in only one place, so I think this is something we actually want to push for as an engineering team. so I am a bit hesitant to ask them if they want this. if they did have a need for it, I don't think there is actually anything stopping us from adding, say, to the script_level's seeding_key to enable this. so that makes me want to just not support it now, and add it later if we need it.

Reply:
That all sounds good to me!

Reply:
ok, great! The Jira item for the follow up which I think we should really do soon is linked above, and I've updated the comment here to reflect the two purposes of seed_key.